### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

**Comment by prolic** [comment]
ping @codeliner

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  - DOCTRINE_VERSION=^2.5 ES_VERSION=dev-master
+
+before_script:
+  - composer self-update
+  - composer require doctrine/dbal:${DOCTRINE_VERSION} --dev --prefer-source
+  - composer require prooph/event-store:${ES_VERSION} --dev --prefer-source
+  - composer install --prefer-source
+
+before_script:
+  - echo "extension = mongo.so" >> ~/.phpenv/versions/$(phpenv version-name)/etc/php.ini
```
Comment (codeliner) [inline .travis.yml]: not needed for doctrine adapter

File: README.md [hunk 2]
```diff
@@ -1,2 +1,2 @@
-# snapshot-doctrine-adapter
+# snapshot-mongodb-adapter
```
Comment (codeliner) [inline README.md]: s/mongodb/doctrine

File: tests/DoctrineSnapshotAdapterTest.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            'dbname' => ':memory:'
+        ];
+
+        $this->connection = DriverManager::getConnection($connection);
+
+        $schema = new Schema();
+        $table = $schema->createTable('foo_snapshot');
+        $table->addColumn('aggregate_type', 'string');
+        $table->addColumn('aggregate_id', 'string');
+        $table->addColumn('last_version', 'integer');
```
Comment (codeliner) [inline tests/DoctrineSnapshotAdapterTest.php]: SnapshotSchema helper would be nice like we have for the event store doctrine adpater: https://github.com/prooph/event-store-doctrine-adapter/blob/master/src/Schema/EventStoreSchema.php

File: tests/DoctrineSnapshotAdapterTest.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+        $readSnapshot = $this->adapter->get($aggregateType, 'id');
+
+        $this->assertEquals($snapshot, $readSnapshot);
+    }
+
+    /**
+     * @test
+     */
+    public function it_uses_custom_snapshot_grid_fs_map_and_write_concern()
```
Comment (codeliner) [inline tests/DoctrineSnapshotAdapterTest.php]: copy & paste method name from mongodb adapter? ;)

**Comment by codeliner** [comment]
@prolic some minor notes but overall great work :+1:

**Comment by prolic** [comment]
@codeliner done.

**Pull request merged**